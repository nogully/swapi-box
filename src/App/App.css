.App {
  text-align: center;
  
}

img {
  width: 400px;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

button {
  display: inline-block;
  border: 1px solid #2d2d2d;
  border-radius: 20px;
  font-size: 14px;
  width: 90px;
  padding: 10px 15px;
  margin: 10px;
}

button:hover {
  background-color: #EEE;
  box-shadow: 2px 2px 5px purple;
}

.dark {
  background-color: purple;
  color: white;
}

.light {
  background-color: #CCC;
  color: #2d2d2d;
  font-weight: bold;
}


@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

