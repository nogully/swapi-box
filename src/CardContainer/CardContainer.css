.CardContainer {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  max-width: 1000px;
  min-width: 1000px;
}