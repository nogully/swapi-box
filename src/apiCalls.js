/* eslint-disable camelcase */
export const resolveEndpoint = async (url) => {
  try {
    const response = await fetch(url);
    const parsed = await response.json();
    if (response.status <= 200) {
      return parsed;
    } else {
      throw new Error("Error in resolveEndpoint");
    }
  } catch (error) { 
    throw new Error("Error in resolveEndpoint");
  }
};

export const getFilmCrawl = async (randomNumber) => {
  try {
    const response = await fetch('https://swapi.co/api/films/');
    const films = await response.json();
    if (response.status <= 200) {
      const { title, episode_id, opening_crawl } = films.results[randomNumber];
      const randomFilm = Object.assign( {}, 
        {title}, 
        {episode_id}, 
        {opening_crawl} );
      return randomFilm;
    } else {
      throw new Error("Error in getFilmCrawl");
    }
  } catch (error) { 
    throw new Error("Error in getFilmCrawl");
  }
};

export const getRandomInt = () => {
  return Math.floor(Math.random() * Math.floor(6));
};

export const getPeople = async () => {
  try {
    const response = await fetch('https://swapi.co/api/people');
    const people = await response.json();
    if (response.status <= 200) {
      return cleanPeople(people.results);
    } else {
      throw new Error("Error in getPeople");
    }
  } catch (error) { 
    throw new Error("Error in getPeople");
  }
};

const cleanPeople = (people) => {
  const unresolvedPromises = people.map( async (person) => {
    const { name, homeworld, species } = person;
    const homeworldData = await resolveEndpoint(homeworld);
    const speciesData = await resolveEndpoint(species);
    const cleaned = Object.assign({}, 
      {name}, 
      {homeworld: homeworldData.name}, 
      {species: speciesData.name}, 
      {population: homeworldData.population}
    );
    return cleaned;
  });
  return Promise.all(unresolvedPromises);
};

export const getPlanets = async () => {
  try {
    const response = await fetch('https://swapi.co/api/planets/');
    const planets = await response.json();
    if (response.status <= 400) {
      return cleanPlanets(planets.results);
    } else { 
      throw new Error("Error in getPlanets");
    }
  } catch (error) { 
    throw new Error("Error in getPlanets");
  }
};

const cleanPlanets = async (planets) => {
  const unresolvedPromises = planets.map( async (planet) => {
    const { name, terrain, population, climate, residents } = planet;
    const residentData = await cleanResidents(residents);
    const cleaned = Object.assign({}, 
      {name}, 
      {terrain}, 
      {climate}, 
      {population}, 
      {residents: residentData.join(', ') }
    );
    return cleaned;
  });
  return Promise.all(unresolvedPromises);
};

const cleanResidents = async (residents) => {
  const unresolvedPromises = residents.map( async (resident) => {
    const person = await resolveEndpoint(resident);
    return person.name;
  });
  return Promise.all(unresolvedPromises);
};

export const getVehicles = async () => {
  try { 
    const vehicles = await resolveEndpoint('https://swapi.co/api/vehicles/');
    const cleaned = await cleanVehicles(vehicles.results);
    return cleaned;
  } catch (error) { 
    return "Error";
  }
};

export const cleanVehicles = async (vehicles) => {
  const unresolvedPromises = vehicles.map( async (vehicle) => {
    const { name, model, vehicle_class, passengers } = vehicle;
    return Object.assign({}, 
      {name}, 
      {model}, 
      {vehicle_class},
      {passengers});
  });
  return Promise.all(unresolvedPromises);
};
