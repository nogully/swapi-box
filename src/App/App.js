import React, { Component } from 'react';
import Header from '../Header/Header.js'
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="div">
        <Header />
        <h1 className="App-title">Explore Star Wars</h1>
        <p className="App-intro"> </p>
      </div>
    );
  }
}

export default App;
