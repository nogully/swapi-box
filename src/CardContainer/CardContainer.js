import React from 'react';
import Card from '../Card/Card'
import './CardContainer.css'

const CardContainer = (props) => {
  var cards = []
  switch (props.category) {
    case 'people': 
      cards = props.cardArray.map((person, index) => {
        const isActive = props.favorites.includes(person.name) ? 'active' : ''
        return (
          <Card name={ person.name } 
                homeworld={ person.homeworld } 
                species={ person.species } 
                population={ person.population }
                favoriteCard={props.favoriteCard}
                active={isActive}
                key={ person.name } />
          ) 
      })
      break;
    case 'planets':
      cards = props.cardArray.map((planet, index) => {
        const isActive = props.favorites.includes(planet.name) ? 'active' : ''
        return (
          <Card name={ planet.name } 
                species={ planet.species } 
                population={ planet.population }
                residents={ planet.residents }
                climate={ planet.climate }
                terrain={ planet.terrain }
                favoriteCard={props.favoriteCard}
                active={isActive}
                key={ planet.name } />
          ) 
      })
      break;
    default:
     console.log('CardContainer switch case hates you');
     return null;
  }

  return (
    <div class="CardContainer">
      { cards }
    </div>
  )

}

export default CardContainer;