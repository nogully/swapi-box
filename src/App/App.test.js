/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import * as apiCalls from '../apiCalls'
import CardContainer from '../CardContainer/CardContainer'


describe('App', () => {
  let wrapper
  let mockPerson = {
          name: "Leia Organa",
          homeworld: "https://swapi.co/api/planets/1/",
          species: "Human",
          population: "https://swapi.co/api/planets/1/"
        }
  let mockFilm = {
          title: "A New Hope"
        }

  window.fetch = jest.fn().mockImplementation( () => {
    return Promise.resolve({
      status: 200, 
      json: () => Promise.resolve( {
        status: 200,
        value:  [ { title: 'A New Hope'} ]
      })
    })
  })

  beforeEach(() => {
    wrapper = shallow(<App />);
    const expectedParams = 'https://swapi.co/api/people'
  })

  it('exists and matches snapshot', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('should start with an empty favorites state', () => {
    expect(wrapper.state('favorites')).toEqual([])
  })

  it('should start with a randomFilm state object from which to draw the crawl', () => {
    // expect(wrapper.state('randomFilm').title).toEqual('A New Hope')
  })

  it.skip('on button click (people, planets, vehicles), resets the state with the corresponding array after adding a category', async () => {
   
  })

  it.skip('calls fetch with the correct params', () => {
   
  })
  
  it.skip('populates the CardContainer with the correct kind of cards', async () => {
 
  })

  it.skip('on click of Favorite button, adds Favorited cards to the state.favorites array', async () => {
 
  })

  it.skip('on 2nd click of Favorite button, removes un-Favorited cards from the state.favorites array', async () => {

  })
})