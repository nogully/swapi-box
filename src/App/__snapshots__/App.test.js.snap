// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should exist and match snapshot 1`] = `
<div
  className="div"
>
  <Header
    displayFavorites={[Function]}
    fetchData={[Function]}
  />
  <Welcome
    randomFilm={Object {}}
  />
</div>
`;
